tools/include/edm4hep2json.hxx

```diff
@@ -152,38 +137,96 @@ std::vector<std::string> splitString(const std::string& inString) {
 }
 
 template <typename ReaderT>
-void read_events(const std::string& filename, const std::string& jsonFile, const std::string& requestedCollections,
-                 int nEventsMax = -1, bool verboser = false) {
+int read_frames(const std::string& filename, const std::string& jsonFile, const std::string& requestedCollections,
+                const std::string& requestedEvents, const std::string& frameName, int nEventsMax = -1,
+                bool verboser = false) {
   ReaderT reader;
   reader.openFile(filename);
 
-  auto store = podio::EventStore();
-  store.setReader(&reader);
-
   nlohmann::json allEventsDict;
 
-  auto collList = splitString(requestedCollections);
-
-  if (verboser) {
-    printCollTypes(store, collList);
-  }
-
-  unsigned nEvents = reader.getEntries();
+  unsigned nEvents = reader.getEntries(frameName);
   if (nEventsMax > 0) {
     if ((unsigned)nEventsMax < nEvents) {
       nEvents = nEventsMax;
     }
   }
-  for (unsigned i = 0; i < nEvents; ++i) {
-    if (verboser && i % 1000 == 0) {
-      std::cout << "INFO: Reading event " << i << std::endl;
+
+  auto collList = splitString(requestedCollections);
+
+  std::vector<int> eventVec;
+  if (!requestedEvents.empty()) {
+    auto eventList = splitString(requestedEvents);
+    for (auto& evnt : eventList) {
+      if (eventVec.size() >= nEvents) {
+        break;
+      }
+
+      int evntNum = -1;
+      try {
+        evntNum = std::stoi(evnt);
+      } catch (...) {
+        if (verboser) {
+          std::cout << "WARNING: This is not a number:\n"
+                    << "         " << evnt << "\n";
+        }
+        continue;
+      }
+
+      if (evntNum < 0) {
+        if (verboser) {
+          std::cout << "WARNING: Event number lower than zero:\n"
+                    << "         " << evnt << "\n";
+        }
+        continue;
+      }
+
+      if ((unsigned)evntNum > nEvents) {
+        if (verboser) {
+          std::cout << "WARNING: Event number larger than number of events in "
+                       "the file or maximal event number to be processed:\n"
+                    << "         " << evnt << "\n";
+        }
+        continue;
+      }
+
+      eventVec.emplace_back(evntNum);
+    }
+
+    if (eventVec.empty()) {
+      return EXIT_SUCCESS;
+    }
+
+    if (verboser) {
+      std::cout << "INFO: Converting the following events:\n";
+      std::string evntStr;
+      for (auto& evnt : eventVec) {
+        evntStr += std::to_string(evnt);
+        evntStr += ",";
+      }
+      evntStr.pop_back();
+      std::cout << "      " << evntStr << std::endl;
```

Comment:
```suggestion
      std::cout << eventVec[0];
      for (size_t i = 1; i < eventVec.size(); ++i) {
        std::cout << "," << eventVec[i];
      }
      std::cout << std::endl;
```
Avoids copying and re-allocating a string each iteration. (Probably not really a problem here since this is only executed once).

Reply:
But what about trailing comma?

Reply:
There is no trailing comma in this case, because we print the first element of the vector and then put a leading comma in front of all other elements.

Reply:
Yes :)